Libraries/LibJS/Runtime/RegExpObject.cpp

```diff
@@ -24,13 +24,81 @@
+    };
+
+    for (auto ch : flags) {
+        switch (ch) {
+        case 'g':
+            if (g)
+                vm.throw_exception<JS::SyntaxError>(global_object, JS::ErrorType::RegExpObjectRepeatedFlag, ch);
+            g = true;
+            options.effective_flags |= regex::ECMAScriptFlags::Global;
+            options.declared_flags |= regex::ECMAScriptFlags::Global;
+            break;
+        case 'i':
+            if (i)
+                vm.throw_exception<JS::SyntaxError>(global_object, JS::ErrorType::RegExpObjectRepeatedFlag, ch);
+            i = true;
+            options.effective_flags |= regex::ECMAScriptFlags::Insensitive;
+            options.declared_flags |= regex::ECMAScriptFlags::Insensitive;
+            break;
+        case 'm':
+            if (m)
+                vm.throw_exception<JS::SyntaxError>(global_object, JS::ErrorType::RegExpObjectRepeatedFlag, ch);
+            m = true;
+            options.effective_flags |= regex::ECMAScriptFlags::Multiline;
+            options.declared_flags |= regex::ECMAScriptFlags::Multiline;
+            break;
+        case 's':
+            if (s)
+                vm.throw_exception<JS::SyntaxError>(global_object, JS::ErrorType::RegExpObjectRepeatedFlag, ch);
+            s = true;
+            options.effective_flags |= regex::ECMAScriptFlags::SingleLine;
+            options.declared_flags |= regex::ECMAScriptFlags::SingleLine;
+            break;
+        case 'u':
+            if (u)
+                vm.throw_exception<JS::SyntaxError>(global_object, JS::ErrorType::RegExpObjectRepeatedFlag, ch);
+            u = true;
+            options.effective_flags |= regex::ECMAScriptFlags::Unicode;
+            options.declared_flags |= regex::ECMAScriptFlags::Unicode;
+            break;
+        case 'y':
+            if (y)
+                vm.throw_exception<JS::SyntaxError>(global_object, JS::ErrorType::RegExpObjectRepeatedFlag, ch);
+            y = true;
+            // Now for the more interesting flag, 'sticky' actually unsets 'global', part of which is the default.
+            options.effective_flags.reset_flag(regex::ECMAScriptFlags::Global);
+            // "What's the difference between sticky and global, then", that's simple.
+            // all the other flags imply 'global', and the "global" flag implies 'stateful';
+            // however, the "sticky" flag does *not* imply 'global', only 'stateful'.
+            options.effective_flags |= (regex::ECMAScriptFlags)regex::AllFlags::Internal_Stateful;
+            options.effective_flags |= regex::ECMAScriptFlags::Sticky;
+            options.declared_flags |= regex::ECMAScriptFlags::Sticky;
+            break;
+        default:
+            vm.throw_exception<JS::SyntaxError>(global_object, JS::ErrorType::RegExpObjectBadFlag, ch);
+            return options;
+        }
+    }
+
+    return options;
+}
+
 namespace JS {
 
 RegExpObject* RegExpObject::create(GlobalObject& global_object, String content, String flags)
```

Comment:
I know you didn't write this bit, but would you agree we rename "content" in every RegExp context (ctor, obj, proto) to "pattern"? That's what the spec and probably most people use for talking about "the string that makes up the regular expression".

Reply:
thanks! :^)